core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -897,14 +902,54 @@ public void start() {
          displayName = "Stops cache."
    )
    public void stop() {
-      stop(null);
+      performImmediateShutdown();
    }
 
-   void stop(ClassLoader explicitClassLoader) {
-      if (log.isDebugEnabled()) log.debugf("Stopping cache %s on %s", getName(), getCacheManager().getAddress());
+   @Override
+   @ManagedOperation(
+         description = "Shuts down the cache",
+         displayName = "Shuts dow the cache"
+   )
+   public void shutdown() {
+      if (log.isDebugEnabled())
+         log.debugf("Shutting down cache %s on %s", getName(), getCacheManager().getAddress());
+      if (!stopping) {
+         synchronized (this) {
+            // DCL to make sure that only one thread calls stop at one time,
+            // and any other calls by other threads are ignored.
+            if (!stopping && componentRegistry.getStatus() == ComponentStatus.RUNNING) {
+               stopping = true;
+               requestGracefulShutdown();
+            }
+         }
+      }
+   }
+
+   private void requestGracefulShutdown() {
+      // If the cache is clustered, perform a cluster-wide shutdown, otherwise do it immediately
+      if (config.clustering().cacheMode().isClustered()) {
+         try {
+            localTopologyManager.cacheShutdown(name);
+         } catch (Exception e) {
+            throw new CacheException(e);
+         }
+      }
+      performImmediateShutdown();
+   }
+
+   private void performImmediateShutdown() {
+      if (log.isDebugEnabled())
+         log.debugf("Stopping cache %s on %s", getName(), getCacheManager().getAddress());
       componentRegistry.stop();
    }
 
+   public void performGracefulShutdown() {
+      // Perform any orderly shutdown operations here
+      Optional<PassivationManager> passivationManager = componentRegistry.getOptionalComponent(PassivationManager.class);
```

Comment:
The `PassivationManager` doesn't seem optional to me, it's always created by `EmptyConstructorNamedCacheFactory`. True, it may not be created if no other component depends on it, but the `getOptionalComponent()` method name gives the wrong impression. And TBH I like the lambda-less version better anyway :)